custom/ucr_ext/expressions.py

```diff
@@ -0,0 +1,431 @@
+from corehq.apps.userreports.expressions.factory import ExpressionFactory
+from jsonobject.base_properties import DefaultProperty
+from corehq.apps.userreports.specs import TypeProperty
+from corehq.apps.userreports.expressions.getters import transform_date
+from dimagi.ext.jsonobject import JsonObject, ListProperty, StringProperty
+
+
+NUM_FUTURE_MONTHS = 3
+
+
+CUSTOM_UCR_EXPRESSIONS = [
+    ('ext_diff_calendar_months', 'custom.ucr_ext.expressions.diff_calendar_months'),
+    ('ext_root_property_name', 'custom.ucr_ext.expressions.root_property_name'),
+    ('ext_iterate_from_opened_date', 'custom.ucr_ext.expressions.iterate_from_opened_date'),
+    ('ext_month_start', 'custom.ucr_ext.expressions.month_start'),
+    ('ext_month_end', 'custom.ucr_ext.expressions.month_end'),
+    ('ext_parent_id', 'custom.ucr_ext.expressions.parent_id'),
+    ('ext_open_in_month', 'custom.ucr_ext.expressions.open_in_month'),
+    ('ext_get_case_forms_by_date', 'custom.ucr_ext.expressions.get_case_forms_by_date'),
+]
+
+
+class DiffCalendarMonthsExpressionSpec(JsonObject):
+    type = TypeProperty('ext_diff_calendar_months')
+    from_date_expression = DefaultProperty(required=True)
+    to_date_expression = DefaultProperty(required=True)
+
+    def configure(self, from_date_expression, to_date_expression):
+        self._from_date_expression = from_date_expression
+        self._to_date_expression = to_date_expression
+
+    def __call__(self, item, context=None):
+        from_date_val = transform_date(self._from_date_expression(item, context))
+        to_date_val = transform_date(self._to_date_expression(item, context))
+        if from_date_val is not None and to_date_val is not None:
+            return ((to_date_val.year - from_date_val.year) * 12) + to_date_val.month - from_date_val.month
+        return None
+
+
+class RootPropertyNameSpec(JsonObject):
+    type = TypeProperty('ext_root_property_name')
+    property_name = StringProperty(required=True)
+    datatype = StringProperty(required=False)
+
+
+class IterateFromOpenedDateSpec(JsonObject):

```

Comment:
i can't tell how these classes that are just wrappers around the type are adding value. can they be removed?